CMake/AbseilHelpers.cmake

```diff
@@ -40,9 +44,12 @@ set(ABSL_IDE_FOLDER Abseil)
 # TESTONLY: When added, this target will only be built if user passes -DABSL_RUN_TESTS=ON to CMake.
 #
 # Note:
-# By default, absl_cc_library will always create a library named absl_internal_${NAME},
-# and alias target absl::${NAME}.
+# By default, absl_cc_library will always create an alias target absl::${NAME}.
+# When ABSL_ENABLE_INSTALL is OFF, target name is absl_${NAME} for PUBLIC libraries
+# and absl_internal_${NAME} for private ones.
+# When ABSL_ENABLE_INSTALL is ON, target name is ${NAME}.
```

Comment:
Personally, I would prefer target to still have `absl_` or `absl::` prefix even when install is enabled. Namespace is always useful.

If an external project intends to use Abseil without install, it needs to depend on Abseil targets as `absl::TARGET` in `CMakeFiles.txt`. However, if user want to build the project with install, CMake won't be able to find the Abseil targets since only `TARGET` is created.

Reply:
Made some mistakes in my last comment, please ignore.

Due to the line `add_library(absl::${ABSL_CC_LIB_NAME} ALIAS ${_NAME})`, `absl::NAME` is always created when `PUBLIC` is set regardless of `ABSL_ENABLE_INSTALL`. So as long as user only depend on `absl::NAME`, Abseil should still build.

But I still don't see why we should strip `absl_[internal_]` prefix when `ABSL_ENABLE_INSTALL` is set.

Reply:
I did not know notice about line 114 (my main concern was the possible collision due to `libfoo.so` naming). Everything makes sense now. Thank you for your detailed explanation.